src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -80,15 +80,12 @@ internal sealed class State
 
             public void RemoveStream(MsQuicStream? stream)
             {
-                bool releaseHandles;
-                lock (this)
-                {
-                    StreamCount--;
-                    Debug.Assert(StreamCount >= 0);
-                    releaseHandles = _closing && StreamCount == 0;
-                }
+                int count = Interlocked.Decrement(ref StreamCount);
+                bool closing = Volatile.Read(ref _closing);
```

Comment:
Why does this need to read _closing?

Reply:
I can move to the `Debug.Assert` bellow. This is only one reason why this is here.